statsd/worker.go

```diff
@@ -0,0 +1,80 @@
+package statsd
+
+import (
+	"math/rand"
+	"sync"
+)
+
+type worker struct {
+	pool   *bufferPool
+	buffer *statsdBuffer
+	sender *sender
+	sync.Mutex
+}
+
+func newWorker(pool *bufferPool, sender *sender) *worker {
+	return &worker{
+		pool:   pool,
+		sender: sender,
+		buffer: pool.borrowBuffer(),
+	}
+}
+
+func (w *worker) processMetric(m metric) error {
+	if w.shouldSample(m.rate) {
```

Comment:
I believe this should be `if !w.shouldSample(m.rate) {}`?

Reply:
The logic is right but the naming is confusing, editing.